### PR TITLE
Convert all tests to .mjs format

### LINKED ISSUE #5540: Convert all test files to .mjs
Broken out of #5531 

We want to make clear that our unit tests are acceptable as non-TypeScript files. 

- [ ] convert all `.test.tsx` files to `.mjs`
- [ ] strip out any typing present in the test files
- [ ] ensure TypeScript intellisense from built code remains working
- [ ] Document this pattern in the [COLLABORATING](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#unit-tests-and-storybooks)

### PR DESCRIPTION
**Pull request by bmuenzenmeyer**
## Description

- Renames all `test.ts{x}` files to `.mjs`
- Removes typing from renamed files
- Adds note to the COLLABORATOR guide
- Does not change storybooks. Open to feedback on that.

## Validation

- ✅ `npx run turbo test`
- ✅ `npx run turbo format`
- ✅ `npx run turbo lint`
- ✅  Do tests benefit from Typescript types? yes

## Related Issues

closes #5540

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing, and/or `npx turbo test:snapshot` to update snapshots if I created and/or updated React Components.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Comment by bmuenzenmeyer** [comment]
@ovflowd this is what the tests look like converted. the intellisense could be better, IMO

**Comment by ovflowd** [comment]
@bmuenzenmeyer issue is the `@` imports in the tests. Can you replace the `@` imports into regular imports? (only in the unit tests) (relative imports), I think we can skip those there and ESM doesn't know what they are.

image

File: hooks/__tests__/useClickOutside.test.mjs [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import { render, fireEvent, screen } from '@testing-library/react';
 import { useState } from 'react';
-import { useClickOutside } from '@/hooks/useClickOutside';
+import { useClickOutside } from '../useClickOutside';
```
Comment (bmuenzenmeyer) [inline hooks/__tests__/useClickOutside.test.mjs:3]: `/hooks` does not have an index, so we have to go up and into the actual file

File: turbo.json [hunk 2]
```diff
@@ -163,7 +163,7 @@
     },
     "test:unit": {
       "inputs": [
-        "{app,components,hooks,i18n,layouts,middlewares,pages,providers,types,util}/**/*.{ts,tsx}",
+        "{app,components,hooks,i18n,layouts,middlewares,pages,providers,types,util}/**/*.{ts,tsx,mjs}",
```
Comment (bmuenzenmeyer) [inline turbo.json:166]: @ovflowd I think I did this correctly, would appreciate a second set of eyes. without this, turbo kept thinking the tests were unchanged

**Review by bmuenzenmeyer — COMMENTED** [review]
self-review, cc @ovflowd for a question

**Review by ovflowd — APPROVED** [review]
LGTM!

**Pull request merged**